packages/babel-plugin-transform-typescript/src/index.js

```diff
@@ -94,6 +135,26 @@ export default declare((api, { jsxPragma = "React" }) => {
         }
       },
 
+      ExportNamedDeclaration(path, { exportableTSNames }) {
+        // remove export declaration if it's exporting only types
+        if (
+          path.node.specifiers.length > 0 &&
+          !path.node.specifiers.find(
+            exportSpecifier =>
+              exportableTSNames.indexOf(exportSpecifier.local.name) === -1,
```

Comment:
1) `exportableTSNames` should probably be a `new Set()`?
2) I don't remember if `@babel/traverse`'s scope tracking supports TypeScript. If it does, we could check `path.sope.getBinding(exportSpecifier.local.name)` instead of using `exportableTSNames`.

Reply:
`scope.getBinding` doesn't seem to work for TS type declarations